mypy_django_plugin/lib/helpers.py

```diff
@@ -118,10 +118,37 @@ def __call__(self, ctx: DynamicClassDefContext) -> None:
         self.semanal_api = cast(SemanticAnalyzer, ctx.api)
         self.create_new_dynamic_class()
 
+    def generate_manager_info_and_module(self, base_manager_info: TypeInfo) -> Tuple[TypeInfo, MypyFile]:
+        new_manager_info = self.semanal_api.basic_new_typeinfo(
+            self.class_name,
+            basetype_or_fallback=Instance(
+                base_manager_info,
+                [AnyType(TypeOfAny.unannotated)])
+        )
+        new_manager_info.line = self.call_expr.line
+        new_manager_info.defn.line = self.call_expr.line
+        new_manager_info.metaclass_type = new_manager_info.calculate_metaclass_type()
+
+        current_module = self.semanal_api.cur_mod_node
+        current_module.names[self.class_name] = SymbolTableNode(
+            GDEF,
+            new_manager_info,
+            plugin_generated=True)
+        return new_manager_info, current_module
+
     @abstractmethod
     def create_new_dynamic_class(self) -> None:
         raise NotImplementedError
 
+class DynamicClassFromMethodCallback(DynamicClassPluginCallback):
+
```

Comment:
Add class attribute `callee` here, it should be certain that it's a `MemberExpr`. Also, add assert with the type in the `__init__` so that it would fail if this subclass is used incorrectly. It will allow a bit nicer autocompletion and maybe will save you an assert somewhere in subclasses.

Reply:
Adding an assert in `__init__` makes like no sense, we don't have `DynamicClassDefContext` available up there (only `NewSemanalDjangoPlugin` instance passed in `ManagerFromQuerySetCallback(self)` in `get_dynamic_class_hook`. I will add an assert in `__call__` though :)

Reply:
Yes, in `__call__`, my bad:)